### PR TITLE
feature(format): Update to use plain functions

### PR DESCRIPTION
**Pull request by c-frater**
## Description

Deprecated:
- hx.format.si(sf, strict): (number): String
- hx.format.exp(sf, strict): (number): String
- hx.format.fixed(digits, strict): (number): String
- hx.format.round(sf, strict): (number): String
- hx.format.zeroPad(length, strict): (number): String

Added: 
- hx.si(number, { sf, strict }): String
- hx.exp(number, { sf, strict }): String
- hx.fixed(number, { digits, strict }): String
- hx.round(number, { sf, strict }): String
- hx.zeroPad(number, { length, strict }): String

## Motivation and Context

First stage of #457 (to be completed in 2.0.0 release)

## How Has This Been Tested?

Added tests for `zeroPad` as they were missing and added tests for the new functions.

## Checklist:

- [x] I have added a tag to this pull request that indicates the impact of the change (patch, minor or major)
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly. (This includes updating the changelog).
- [x] I have read the **CONTRIBUTING** document.
- [x] All my changes are covered by tests.
- [x] This request is ready to review and merge

### CODE REVIEW

File: docs/content/docs/format/api/auto.um [hunk 1]
```diff
@@ -134,3 +164,166 @@
...
+  @added nextReleaseVersion
+    @issue 457
+    @description
+      Added plain functions for formatting numbers that take the number and
+      return the formatted number (as opposed to a curried function)
+
+  @description
+    A function to round a number to a specific number of significant figures.
+    @@codeblock js
+      hx.format.round(4)(3.1415926) // Returns "3.142"
```
Comment (jmsmyth) [inline docs/content/docs/format/api/auto.um]: Needs updating to match the new api

Comment (jmsmyth) [reply docs/content/docs/format/api/auto.um]: Same for the other new functions as well

File: modules/format/main/index.coffee [hunk 2]
```diff
@@ -1,62 +1,78 @@
 
 siSuffixes = ['y','z','a','f','p','n','µ','','','K','M','G','T','P','E','Z','Y']
 
-zeroPad = (number, pad) ->
+zeroPad = (number, { length }) ->
```
Comment (jmsmyth) [inline modules/format/main/index.coffee]: Should there be a default for this? Or maybe this one should be non optional?

File: modules/format/main/index.coffee [hunk 3]
```diff
@@ -1,62 +1,78 @@
...
-formatFixed = (n, digits) ->
-  if isNaN(n) then return 'NaN'
-  n.toFixed(digits)
 
-strictCheck = (f, sf, strict) ->
-    if strict
-      (n) -> f(n, sf)
-    else
-      (n) -> if hx.isString(n) then n else f(n, sf)
+hx.round = (number, options) -> strictCheck(number, formatRound, options)
```
Comment (jmsmyth) [inline modules/format/main/index.coffee]: The options objects should have defaults - if options is undefined, these functions will throw an error. Or if the options are required, a better error should be thrown when the options are not provided

**Comment by jmsmyth** [comment]
Also, I think we need to make the 1.x.x branch and merge into that first. Merging this into master directly might be problematic 😁

File: docs/content/docs/format/api/auto.um (hunk 1 above)
Comment (c-frater) [reply docs/content/docs/format/api/auto.um]: Done

File: modules/format/main/index.coffee (hunk 2 above)
Comment (c-frater) [reply modules/format/main/index.coffee]: Decided to go with a default. If this length should be non-optional, the digits/sf should also be non-optional

File: modules/format/main/index.coffee (hunk 3 above)
Comment (c-frater) [reply modules/format/main/index.coffee]: Decided to go with a default of 2 for the digits/sf/length.

**Comment by c-frater** [comment]
I'd already pushed the 1.17.0 branch but forgot to change the target for this PR - I've done that now

**Comment by jmsmyth** [comment]
Do the new functions have any tests?

File: modules/format/test/spec.coffee [hunk 4]
```diff
@@ -1,139 +1,336 @@
...
+    check hx.format.zeroPad(5)(123), '00123'
+    check hx.format.zeroPad(5)(1234), '01234'
+    check hx.format.zeroPad(5)(12345), '12345'
+    check hx.format.zeroPad(5)(123456), '123456'
+
+    check hx.format.zeroPad(5)('string test'), 'string test'
+    check hx.format.zeroPad(5, false)('string test'), 'string test'
+    check hx.format.zeroPad(5, true)('string test'), 'NaN'
+
+  it 'round works',  ->
```
Comment (c-frater) [inline modules/format/test/spec.coffee]: New function tests :)

Probably missed because the file was 'collapsed'

**Pull request merged**